### PR TITLE
Add support for fiction.live (closes #201)

### LINKED ISSUE #201: Requesting support for https://fiction.live/

### PR DESCRIPTION
**Pull request by HazelSh**
I've written a fiction.live adapter for fanficfare. It's been a bit of a mission: the site is very dynamic and the api isn't documented. Closes #201 which is the enhancement-request to support this site. 

I don't have any automated tests, but I've been using it to follow fiction.live stories and have ~20 stories that it's been manually tested with, so I'm confident that it does work.

### CODE REVIEW

**Comment by JimmXinu** [comment]
Single comment (instead of review) didn't show up here, so I don't know if GitHub alerted you.

Anyway, there's an INI [typo](https://github.com/JimmXinu/FanFicFare/pull/535/files#r475235612).

File: calibre-plugin/plugin-defaults.ini [hunk 1]
```diff
@@ -3299,6 +3299,8 @@ tags_label:Tags
 live_label:Next Live Session
 likes_label:Likes
 reader_input_label:Reader Input
+keep_in_order_tags:trueA
```
[the remark this answers is not in the document]
Comment (HazelSh) [reply calibre-plugin/plugin-defaults.ini]: >_<
 Guess that's what I get for doing rushed late night fixes. In my defence, the tags weren't getting sorted after that.

**Comment by JimmXinu** [comment]
Failed on the first test I tried, using a story picked from the front page:
https://fiction.live/stories/Magical-Girl-Lyrical-Nanoha-RaptorS/nGZdw54rui4RWPx68

Partial log:
```
...
FFF: DEBUG: 2020-08-24 13:49:25,697: configurable.py(1143): #####################################
pagecache(GET) MISS: https://fiction.live/api/anonkun/chapters/nGZdw54rui4RWPx68/1529157959269/1529198000694/
FFF: DEBUG: 2020-08-24 13:49:25,997: configurable.py(1001): Encoding:utf8
FFF: DEBUG: 2020-08-24 13:49:26,305: configurable.py(1143): #####################################
pagecache(GET) MISS: https://ddx5i92cqts4o.cloudfront.net/images/1cg53haqf_1c9bbm4l2_Pixiv.Id.454919.full.1689824.jpg
FFF: DEBUG: 2020-08-24 13:49:26,995: configurable.py(1143): #####################################
pagecache(GET) MISS: https://ddx5i92cqts4o.cloudfront.net/images/1cg5c17ip_1cfnrq1ct_RAPTORSSpecSheet7.png
Traceback (most recent call last):
  File "fanficfare\cli.py", line 620, in <module>
    main()
  File "fanficfare\cli.py", line 307, in main
    do_download(url,
  File "fanficfare\cli.py", line 492, in do_download
    output_filename = write_story(configuration, adapter, options.format,
  File "fanficfare\cli.py", line 87, in write_story
    writer.writeStory(outstream=outstream, metaonly=metaonly)
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\writers\base_writer.py", line 201, in writeStory
    self.story = self.adapter.getStory() # get full story
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\adapters\base_adapter.py", line 237, in getStory
    data = self.getChapterTextNum(url,index)
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\adapters\base_adapter.py", line 389, in getChapterTextNum
    return self.getChapterText(url)
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\adapters\adapter_fictionlive.py", line 240, in getChapterText
    text += handler(chunk)
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\adapters\adapter_fictionlive.py", line 348, in format_choice
    options = self.count_votes(chunk)
  File "C:\Users\username\Desktop\nook\FanFicFare\fanficfare\adapters\adapter_fictionlive.py", line 327, in count_votes
    choices = chunk['choices']
KeyError: 'choices'
```

**Comment by HazelSh** [comment]
ok, looked into it. apparently you don't need to have any vote options in a vote chunk. whoops. 

I did a once-over of my code looking for assumptions about chunk contents after realising just to what degree everything in this api is optional -- I think that was after realising that votes didn't need to have *voters* -- but I missed this one. Done another sanity check for that particular mistake, and confirmed that that story works now. 

I really should do some more automated testing.

**Comment by JimmXinu** [comment]
First test download worked.  

Second also worked, but put a `<br>` tag in the title: https://fiction.live/stories/Third-Option-Alt-Story-br-/s5hnnnCPYdpTTXSW8

Most adapters include `from ..htmlcleanup import stripHTML`.
I suggest calling `stripHTML()` on title and on most `setMetadata()` calls (except for `setDescription()`) if there's a chance it may contain HTML.

**Comment by HazelSh** [comment]
Ok, thanks. I'll see about adding that.

**Comment by JimmXinu** [comment]
Apparently Git doesn't like that I cleaned up some trailing whitespace.  Oh well--the changes in the `master`, so I'm just going to close this PR rather than resolve a bunch of whitespace changes.

Thanks for contributing this.

**Pull request closed without merging**